### ISSUE TITLE
Query Locations for fillfromlocationid choice box instead of caching locations

### ISSUE THREAD

**Issue by kbower** [kbower/tickettest1#1629]
_Imported from trac ticket 1629.
Created by **kbower, mariam**
Opened in trac: May 3, 2012, 4:40:16 PM
Last modified in trac: Oct 31, 2012, 7:54:35 AM_
***
_by kbower, mariam:_
This is the dialog box that appears when you click "Fill from this store".  

In that case, query inventory instead of locations.

The part that makes me uncomfortable about this is that, since this is a store piece, the sales person should already know in which location in the store the piece resides and showing him a list will only propagate inventory errors because the lazy user will just choose whatever location shows up.  It is also the cause of extra database queries.

So, this should be a system parameter defaulted off.

***
* kbower attached [5sgwjlfy.aqm.png](https://trac.retailarchitects.com/trac/attachment/ticket/1629/5sgwjlfy.aqm.png) on Oct 26, 2012, 9:42:49 AM

**Comment by kbower** [comment]
_**kbower** trac comment on Sep 19, 2012, 12:24:21 PM:_

Also, I've already given in to doing this eventually, but Mariam agreed at dinner at the Steinhafels sprint that, while annoying them, this isn't as urgent as being able to close orders and move off GERS.

It's ticket #1629.

I guess I'm glad to hear that was the worst of their problems. ;)

**Comment by kbower** [comment]
_**rneville** trac comment on Oct 25, 2012, 4:02:09 PM:_

I'm basically done with this, however I did not make it a system option. I've done it as a client-side "lazy load" that only fires if you click on the fill from store checkbox. This seems perfectly reasonable to me and should have little impact on performance except when needed. The manual override in the details section remains unchanged so that a manager can pick a location regardless of whether there's inventory in it or not for whatever reason.

**Comment by kbower** [comment]
_**rneville** trac comment on Oct 25, 2012, 4:06:56 PM:_

(Changeset [[3306]](../commit/2d59d055c83d4d7462d0929642a2c192f8798789))
* fixes #1629 (Query Locations for fillfromlocationid choice box instead of caching locations)

**Comment by kbower** [comment]
_**rneville** trac status change on Oct 25, 2012, 4:06:56 PM:_ **closed**

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 26, 2012, 9:42:49 AM:_


Flex crashes when attempt to click unfill for a filled line and then save.

[[Image(5sgwjlfy.aqm.png)]]

**Comment by kbower** [comment]
_**kbower** trac status change on Oct 26, 2012, 9:42:49 AM:_ **reopened**

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 26, 2012, 9:44:59 AM:_

 * Crashes flex (attempt to use Unfill check box)
 * Lazy load code should be auto generated (if lazy load is the appropriate solution)
 * controller query should do one query instead of two
 * Ideally controlled by system parameter to be kind to server requests

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 26, 2012, 9:57:19 AM:_

lazy load properties generally shouldn't be serialized back and forth to/from server (this is what is causing the crash).  For good reason also, since they generally are data you don't want to go through the trouble of querying if possible and often contain substantial data amounts (e.g., customer history).  We probably can't have this live on the OrderLine object as that is too dynamic a beast.
Also, is it smart enough to know it needs to be 'expired' if the product changes, etc.?  So there are problems on the OrderLine, though it seems to be a good candidate otherwise...

**Comment by kbower** [comment]
_**rneville** trac comment on Oct 26, 2012, 10:21:25 AM:_

It would take me twice as long just to understand what to change and it sounds like you could probably fix this in a matter of minutes.

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 27, 2012, 5:11:01 PM:_

Can't create new order:
```
A fatal error was encountered when communicating with the server.

Server Debug is enabled, output can be found here:
https://dev.retailarchitects.com/tg/_debug/view/1351277274

Error message:
APIError: must pass a site id
```

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 27, 2012, 5:12:52 PM:_

(Changeset [[3320]](../commit/73faeb8f9fbe5165e60f2cacf6752bb54f194448))
* need to pull [[3306]](../commit/2d59d055c83d4d7462d0929642a2c192f8798789) back out; breaks writing new orders (see #1629)

**Comment by kbower** [comment]
_**kbower** trac comment on Oct 31, 2012, 7:54:35 AM:_

(Changeset [[3337]](../commit/045ad0d9b01be125d95ec5f25761e099344ff236))
* fixes #1629 (Parameter to Query Locations for fillfromlocationid choice box instead of caching locations)

**Comment by kbower** [comment]
_**kbower** trac status change on Oct 31, 2012, 7:54:35 AM:_ **closed**
